Makefile

```diff
@@ -215,64 +215,73 @@ $(BUILD_OUTPUT_DIRECTORY): | $(BUILD_DIRECTORY)
+	package.json npm-shrinkwrap.json \
 	| $(BUILD_DIRECTORY)
-	mkdir $@
+	mkdir -p $@
 	./scripts/build/dependencies-npm.sh -p \
 		-r "$(TARGET_ARCH)" \
 		-v "$(NODE_VERSION)" \
-		-x $@ \
+		-x $(dir $@) \
 		-t node \
 		-s "$(TARGET_PLATFORM)"
-	git apply --directory $@/node_modules/lzma-native patches/cli/lzma-native-index-static-addon-require.patch
+	git apply --directory $@/lzma-native patches/cli/lzma-native-index-static-addon-require.patch
 
 $(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app: \
-	package.json lib \
-	$(BUILD_DIRECTORY)/node-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies \
-	| $(BUILD_DIRECTORY)
-	mkdir $@
-	cp $(word 1,$^) $@
-	$(CPRF) $(word 2,$^) $@
-	$(CPRF) $(word 3,$^)/* $@
-
-$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH).js: \
-	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app \
+	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/node_modules \
 	| $(BUILD_DIRECTORY)
-	./scripts/build/concatenate-javascript.sh -e lib/cli/etcher.js -b $< -o $@ -m
+	cp -rf lib $@
+	cp -rf package.json $@
 
-$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH): \
-	$(BUILD_DIRECTORY)/node-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies \
-	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH).js \
-	| $(BUILD_DIRECTORY) $(BUILD_TEMPORARY_DIRECTORY)
-	mkdir $@
-	./scripts/build/node-package-cli.sh -o $@ -l $</node_modules \
-		-n $(APPLICATION_NAME) \
-		-e $(word 2,$^) \
-		-r $(TARGET_ARCH) \
-		-s $(TARGET_PLATFORM)
+TARGET_PLATFORM_PKG = $(TARGET_PLATFORM)
+ifeq ($(TARGET_PLATFORM_PKG),darwin)
+TARGET_PLATFORM_PKG = macos
+endif
+ifeq ($(TARGET_PLATFORM_PKG),win32)
+TARGET_PLATFORM_PKG = win
+endif
 
+ETCHER_CLI_BINARY = $(APPLICATION_NAME_LOWERCASE)
 ifeq ($(TARGET_PLATFORM),win32)
-	./scripts/build/electron-brand-exe.sh \
-		-f $@/etcher.exe \
-		-n $(APPLICATION_NAME) \
-		-d "$(APPLICATION_DESCRIPTION)" \
-		-v "$(APPLICATION_VERSION)" \
-		-c "$(APPLICATION_COPYRIGHT)" \
-		-m "$(COMPANY_NAME)" \
-		-i assets/icon.ico \
-		-w $(BUILD_TEMPORARY_DIRECTORY)
+ETCHER_CLI_BINARY = $(APPLICATION_NAME_LOWERCASE).exe
 endif
 
-ifeq ($(TARGET_PLATFORM),darwin)
-ifdef CSC_NAME
-	./scripts/build/electron-sign-file-darwin.sh -f $@/etcher -i "$(CSC_NAME)"
-endif
-endif
+$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH): \
+	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app \
+	| $(BUILD_DIRECTORY) $(BUILD_TEMPORARY_DIRECTORY)
+	mkdir $@
+	$(NPX) pkg --output $@/$(ETCHER_CLI_BINARY) -t node6-$(TARGET_PLATFORM_PKG)-$(TARGET_ARCH) $</lib/cli/etcher.js
```

Comment:
I wonder if there should be a Makefile 'constant' (variable) for this `lib/cli/etcher.js` path?